test/TestHistogram.cc

```diff
@@ -94,22 +151,17 @@ TEST_F(HistogramTest, TestMultithreadingPerformance) {
 
     t.Start("single_threaded");
     carta::Histogram hist_st(1024, 0.0f, 1.0f, data);
-    hist_st.setup_bins();
-    auto results_st = hist_st.GetHistogram();
     t.End("single_threaded");
 
     carta::ThreadManager::SetThreadLimit(4);
     t.Start("multi_threaded");
-    carta::Histogram mt(1024, 0.0f, 1.0f, data);
-    mt.setup_bins();
-    auto results_mt = mt.GetHistogram();
+    carta::Histogram hist_mt(1024, 0.0f, 1.0f, data);
     t.End("multi_threaded");
 
     auto st_time = t.GetMeasurement("single_threaded");
     auto mt_time = t.GetMeasurement("multi_threaded");
     double speedup = st_time / mt_time;
-    cout << speedup << endl;
-    EXPECT_GE(speedup, 1.5);
+    EXPECT_GE(speedup, 1.5) << "Speedup is: " << speedup;
```

Comment:
Definitely work. It will only print if the test fails. I think it's pretty cool, see https://github.com/google/googletest/blob/master/docs/advanced.md#teaching-googletest-how-to-print-your-values for more advanced examples. (TL;DR: Every macro can do that.)